www/js/main.js

```diff
@@ -4625,69 +4625,82 @@ var showHome = ( function() {
                 name = button.siblings( "[id='station_" + id + "']" ),
                 showSpecialOptions = function( value ) {
 					var opts = select.find( "#specialOpts" ),
-						hex = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].sd : "",
-
-						// Remember station type to understand special data format
-						hexFormat  = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].st : 0;
+						data = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].sd : "",
+						type  = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].st : 0;
 
 					opts.empty();
 
-					if ( value === 1 ) {
+					if ( value === 0 ) {
+						opts.append(
+							"<p class='special-desc center small'>" +
+								_( "Select the station type using the dropdown selector above and configure the station properties." ) +
+							"</p>"
+						).enhanceWithin();
+					} else if ( value === 1 ) {
+						data = ( type === value ) ? data : "0000000000000000";
+
 						opts.append(
 							"<div class='ui-bar-a ui-bar'>" + _( "RF Code" ) + ":</div>" +
-							"<input class='center' data-corners='false' data-wrapper-class='tight ui-btn stn-name' id='rf-code' type='text' value='" + hex + "'>"
+							"<input class='center' data-corners='false' data-wrapper-class='tight ui-btn stn-name' id='rf-code' required='true' type='text' value='" + data + "'>"
 						).enhanceWithin();
 					} else if ( value === 2 ) {
-						var data = parseRemoteStationData( hex );
+						data = ( type === value ) ? parseRemoteStationData( data ) : parseRemoteStationData( "00000000005000" );
```

Comment:
Wow, you are super quick !!!
I just finished a minor commit to clear up a jshint error and I think you are right, your suggestion is clearer. I will create a new commit for this now. I will do the Firmware pull in the next few minutes.
Cheers, Pete
      From: Samer Albahra <notifications@github.com>
 To: OpenSprinkler/OpenSprinkler-App <OpenSprinkler-App@noreply.github.com> 
Cc: PeteBa <peter_basham@yahoo.co.uk>; Author <author@noreply.github.com>
 Sent: Saturday, 4 June 2016, 16:53
 Subject: Re: [OpenSprinkler/OpenSprinkler-App] Added support for HTTP Stations (#32)
   
In www/js/main.js:>  						).enhanceWithin();
>  					} else if ( value === 2 ) {
> -						var data = parseRemoteStationData( hex );
> +						data = ( type === value ) ? parseRemoteStationData( data ) : parseRemoteStationData( "00000000005000" );
Sorry if it's being nit picky but do you think this is easier to read?data = parseRemoteStationData( type === value ? data : "00000000005000" );—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.